segmentstore/contracts/src/main/java/io/pravega/service/contracts/AttributeUpdate.java

```diff
@@ -25,10 +26,40 @@
 @EqualsAndHashCode(of = {"attributeId", "value"})
 @NotThreadSafe
 public class AttributeUpdate {
+    /**
+     * The ID of the Attribute to update.
+     */
     private final UUID attributeId;
+
+    /**
+     * The UpdateType of the attribute.
+     */
     private final AttributeUpdateType updateType;
+
+    /**
+     * The new Value of the attribute.
+     */
     private long value;
 
+    /**
+     * If UpdateType is ReplaceIfEquals, then this is the value that the attribute must currently have before making the
+     * update. Otherwise this field is ignored.
+     */
+    private final long comparisonValue;
```

Comment:
Just to clarify: we don't want the semantic of replaceIfGreater to by replace with value if comparisonValue > previous value. (Nor do we want any form of >= comparison).